pkg/sqlite/database_postgres.go

```diff
@@ -0,0 +1,90 @@
+package sqlite
+
+import (
+	"fmt"
+
+	"github.com/doug-martin/goqu/v9"
+	_ "github.com/doug-martin/goqu/v9/dialect/postgres"
+	_ "github.com/jackc/pgx/v5/stdlib"
+	"github.com/jmoiron/sqlx"
+	"github.com/stashapp/stash/pkg/logger"
+)
+
+type PostgresDB Database
+
+func NewPostgresDatabase(dbConnector string) *Database {
+	dialect = goqu.Dialect("postgres")
+
+	db := &PostgresDB{
+		storeRepository: newDatabase(),
+		lockChan:        make(chan struct{}, 1),
```

Comment:
The reason we have `lockChan` is due to the fact that SQLite can't perform multiple queries in parallel. This is not a problem for Postgres and I'd strongly recommend not using locks in Stash to ensure a single query at the same time, or it would defeat most of the perf benefits of using Postgres.

In the implementation, this code: https://github.com/NodudeWasTaken/stash/blob/aaaad7916f73306fe6af4c49eea9d3c9a22f96a4/pkg/sqlite/database.go#L256-L270

Should just be a no-op when using Postgres (no locks acquired, nor released)

PS: I also think we should not use locks for SQLite too. Instead, we should open the SQLite connection with a `busy_timeout(5000)` parameter added to the query string, which causes SQLite to wait if the DB is busy. This will make the need for these locks redundant.

Reply:
Rewriting the sqlite locks is beyond the scope of this pr, but i have removed the locks for postgresql.